operator/src/main/java/oracle/kubernetes/operator/Watcher.java

```diff
@@ -187,14 +187,22 @@ private void handleRegularUpdate(Watch.Response<T> item) {
 
   private void handleErrorResponse(Watch.Response<T> item) {
     V1Status status = item.status;
-    if (status != null && status.getCode() == HTTP_GONE) {
+    if (status == null) {
+      // The kubernetes client parsing logic can mistakenly parse a status as a type
+      // with similar fields, such as V1ConfigMap. In this case, the actual status is
+      // not available to our layer, so respond defensively by resetting resource version.
+      resourceVersion = 0l;
+    } else if (status.getCode() == HTTP_GONE) {
+      resourceVersion = 0l;
       String message = status.getMessage();
       int index1 = message.indexOf('(');
       if (index1 > 0) {
         int index2 = message.indexOf(')', index1 + 1);
         if (index2 > 0) {
           String val = message.substring(index1 + 1, index2);
-          resourceVersion = !isNullOrEmptyString(val) ? Long.parseLong(val) : 0;
+          if (!isNullOrEmptyString(val)) {
+            resourceVersion = Long.parseLong(val);
+          }
         }
```

Comment:
The longer the method, the harder it is for the next person in to understand it without reverse engineering. I think this method has gotten too long at this point. In-line comments can actually aggravate this. We're better off extracting a method. Let me suggest the following:

static final Pattern RESOURCE_PATTERN = Pattern.compile("(.?\\()(\\\\d+)");

private void handleErrorResponse(Watch.Response<T> item) {
  resourceVersion = computeNextResourceVersion(item.status);
    : 

(rest of method)
}

// The kubernetes client parsing logic can mistakenly parse a status as a type
// with similar fields, such as V1ConfigMap. In this case, the actual returned status is
// null, so respond defensively by resetting resource version.
// In addition, the message returned by OpenShift does not include the next valid resource
// in parenthesis, so once again we reset the resource version.
private long computeNextResourceVersion(V1Status status) {
  if (status == null) return 0L;

 Matcher matcher = RESOURCE_PATTERN.matcher(text);
 if (!matcher.find()) return 0L;
else return Long.parseLong(matcher.group(0));
}